client/blocks/signup-form/social.jsx

```diff
@@ -70,8 +71,15 @@ class SocialSignupForm extends Component {
 	}
 
 	render() {
+		const { currentRoute } = this.props;
 		const uxMode = this.shouldUseRedirectFlow() ? 'redirect' : 'popup';
-		const redirectUri = uxMode === 'redirect' ? `https://${ window.location.host }/start` : null;
+		let redirectUri = uxMode === 'redirect' ? `https://${ window.location.host }/start` : null;
+
+		// For the Jetpack Connect-in-place flow - see p1HpG7-7nj-p2 for more information.
+		if ( window && window.opener && '/jetpack/connect/authorize' === currentRoute ) {
```

Comment:
No use to test if `window` exists here since it's used two lines before without the check. :-)

When testing, you should do `undefined !== typeof window` (for reasons I can't now remember) 😅

Reply:
😅 good spot, thanks! This was all pretty hacky anyway 😉